LKExp-mbed-x-nucleo-iks01a2/src/main.cpp

```diff
@@ -0,0 +1,191 @@
+  BSP_LCD_DisplayStringAt(0, 0, (uint8_t*)"IMU example", CENTER_MODE);
+  uint8_t lcd_string[100];
+  BSP_LCD_SetFont(&Font20);
+#endif
+ 
+  while(1) {
+#ifndef STM32F769DISCO
+    printf("\r\n");
+#endif
+    
+    hum_temp->get_temperature(&value1);
+    hum_temp->get_humidity(&value2);
+#ifndef STM32F769DISCO
+    printf("HTS221: [temp] %7s C,   [hum] %s%%\r\n", print_double(buffer1, value1), print_double(buffer2, value2));
+#else
+    sprintf((char*)lcd_string,"HTS221: [temp] %7s C,   [hum] %s%%               ", print_double(buffer1, value1), print_double(buffer2, value2));
+    BSP_LCD_DisplayStringAt(0, 100, (uint8_t*)lcd_string, LEFT_MODE);
+#endif
+
+    press_temp->get_temperature(&value1);
+    press_temp->get_pressure(&value2);
+#ifndef STM32F769DISCO
+    printf("LPS22HB: [temp] %7s C, [press] %s mbar\r\n", print_double(buffer1, value1), print_double(buffer2, value2));
+    printf("---\r\n");
+#else
+    sprintf((char*)lcd_string,"LPS22HB: [temp] %7s C, [press] %s mbar               ", print_double(buffer1, value1), print_double(buffer2, value2));
+    BSP_LCD_DisplayStringAt(0, 150, (uint8_t*)lcd_string, LEFT_MODE);
+#endif
+    
+    magnetometer->get_m_axes(axes);
+#ifndef STM32F769DISCO
+    printf("LSM303AGR [mag/mgauss]:  %6ld, %6ld, %6ld\r\n", axes[0], axes[1], axes[2]);
+#else
+    sprintf((char*)lcd_string,"LSM303AGR [mag/mgauss]:  %6ld, %6ld, %6ld               ", axes[0], axes[1], axes[2]);
+    BSP_LCD_DisplayStringAt(0, 200, (uint8_t*)lcd_string, LEFT_MODE);
+#endif
+
+    accelerometer->get_x_axes(axes);
+#ifndef STM32F769DISCO
+    printf("LSM303AGR [acc/mg]:  %6ld, %6ld, %6ld\r\n", axes[0], axes[1], axes[2]);
+#else
+    sprintf((char*)lcd_string,"LSM303AGR [acc/mg]:  %6ld, %6ld, %6ld               ", axes[0], axes[1], axes[2]);
+    BSP_LCD_DisplayStringAt(0, 250, (uint8_t*)lcd_string, LEFT_MODE);
+#endif
+
+    acc_gyro->get_x_axes(axes);
+#ifndef STM32F769DISCO
+    printf("LSM6DSL [acc/mg]:      %6ld, %6ld, %6ld\r\n", axes[0], axes[1], axes[2]);
+#else
+    sprintf((char*)lcd_string,"LSM6DSL [acc/mg]: %6ld, %6ld, %6ld               ", axes[0], axes[1], axes[2]);
+    BSP_LCD_DisplayStringAt(0, 300, (uint8_t*)lcd_string, LEFT_MODE);
+#endif
+
+    acc_gyro->get_g_axes(axes);
+#ifndef STM32F769DISCO
+    printf("LSM6DSL [gyro/mdps]:   %6ld, %6ld, %6ld\r\n", axes[0], axes[1], axes[2]);
+#else
+    sprintf((char*)lcd_string,"LSM6DSL [gyro/mdps]: %6ld, %6ld, %6ld               ", axes[0], axes[1], axes[2]);
+    BSP_LCD_DisplayStringAt(0, 350, (uint8_t*)lcd_string, LEFT_MODE);
+#endif
+ 
+    if (mems_event) {
+    mems_event = 0;
+    LSM6DSL_Event_Status_t status;
+    acc_gyro->get_event_status(&status);
+    if (status.WakeUpStatus) {
+        /* Led blinking. */
+        myled = 1;
+  
+        /* Output data. */
+#ifndef STM32F769DISCO
+        printf("Wake Up Detected!\r\n");
+#else
+        BSP_LCD_DisplayStringAt(0, 450, (uint8_t*)"Wake Up Detected!", CENTER_MODE);
+#endif
+        wait_us(3000000);
```

Comment:
si tu veux composer un peu avec mbed pour voir comment ça se débrouille, tu peux utilise `ThisThread::sleep_for()` en ms (je crois)

Reply:
attention parce que wait_us est bloquant (à vérifier) donc si on sleep for, le reste tourne.

mais on peut écrire sur deux lignes (si pas déjà le cas) pour avoir les valeurs en temps réel d'une part et le wakeup ou autre event detected en parallèle

Reply:
J'ai essayé les deux dans [ce commit](https://github.com/leka/LekaOS_Explorations/pull/2/commits/6949ebf4b05344530cc331442b5f2216cc78675c).
Aucun des deux ne semble bloquant, par contre le `ThisThread::sleep_for()`, qui est bien en ms, entraîne des artefacts sur l'écran LCD.
Pour l'interaction avec un event, ça a été testé dans un commit ultérieur, et le résultat est identique : non bloquant.